lls/cache.c

```diff
@@ -351,51 +350,118 @@ xmit_icmp_ping_reply(struct gatekeeper_if *iface, struct rte_mbuf *pkt)
 			RTE_ETHER_TYPE_IPV4);
 	}
 
-	/*
-	 * Set-up IPv4 header.
-	 *
-	 * According to RFC 792 page 13: to form an echo reply
-	 * message, the source and destination addresses are simply reversed,
-	 * the type code changed to 0, and the checksum recomputed.
-	 */
+	/* Set-up IPv4 header. */
 	icmp_ipv4 = (struct rte_ipv4_hdr *)pkt_out_skip_l2(iface, icmp_eth);
+	icmph = (struct rte_icmp_hdr *)ipv4_skip_exthdr(icmp_ipv4);
 
 	icmp_ipv4->time_to_live = IP_DEFTTL;
-	ip_addr_tmp = icmp_ipv4->src_addr;
-	icmp_ipv4->src_addr = icmp_ipv4->dst_addr;
-	icmp_ipv4->dst_addr = ip_addr_tmp;
+	switch (icmph->icmp_type) {
+	case ICMP_ECHO_REQUEST_TYPE: {
+		/*
+		 * According to RFC 792 page 13: to form an echo reply
+		 * message, the source and destination addresses are simply
+		 * reversed, the type code changed to 0, and the checksum
+		 * recomputed.
+		 */
+		uint32_t ip_addr_tmp = icmp_ipv4->src_addr;
```

Comment:
Let's match the type of fields `src_addr` and `dst_addr` replacing `uint32_t` with `rte_be32_t`.